src/test/clojure/xtdb/sql_test.clj

```diff
@@ -2965,3 +2965,22 @@ FROM dates"))))
              (xt/q tu/*node* [(format "FROM (XTQL $$ %s $$) t SELECT *, ? AS x"
                                       (pr-str '#(from :bar [{:xt/id %} *])))
                               2 "x"])))))
+
+(t/deftest temporal-filter-expressions-3306
```

Comment:
worth some unit tests against `->temporal-bounds` here?

Reply:
ha, in fact, we have a test in scan-test with a note saying to adapt for `scan/->temporal-bounds` - good time to do that, while you're in context?

Reply:
Do you want me to add temporal column predicate constraints to `->temporal-bounds` (if possible)?

Reply:
I have added an extra test. The other test is more about pushing temporal predicates into the `temporal-bounds`.